FIPS readme: updates for CNG accuracy, other clarity (#1432)
microsoft/go: eng/doc/fips/README.md

| Build-time config | Runtime config | Internal Microsoft crypto policy | FIPS behavior |
| --- | --- | --- | --- |
| Default | Default | Not compliant | Crypto usage is not FIPS compliant. |
| `GOEXPERIMENT=systemcrypto` | Default | Compliant | Can be used to create a compliant app. FIPS mode is determined by system-wide configuration. Make sure you are familiar with your platform's system-wide FIPS switch, described in [Usage: Runtime](#usage-runtime). |

> | `GOEXPERIMENT=systemcrypto` | `GOFIPS=1` | Compliant | Can be used to create a compliant app. Depending on platform, the app either enables FIPS mode or ensures it is already enabled. If it's not possible, the app panics. See [Usage: Runtime](#usage-runtime). |

Reviewer: ```suggestion
| `GOEXPERIMENT=systemcrypto` | `GOFIPS=1` | Compliant | Can be used to create a compliant app. Depending on platform, the app either enables FIPS mode or ensures it is already enabled. If it's not possible to enable the FIPS mode, then the app panics. See [Usage: Runtime](#usage-runtime). |
```

Author: That's not quite what it means, though--it doesn't just panic if it can't enable FIPS mode, it also panics if it can't ensure that it's enabled. I think I figured out a way to make it more straightforward:

```suggestion
| `GOEXPERIMENT=systemcrypto` | `GOFIPS=1` | Compliant | Can be used to create a compliant app. Depending on platform, the app either enables FIPS mode or ensures it is already enabled. The app panics if there is a problem. See [Usage: Runtime](#usage-runtime). |
```

Versions not listed above are not supported at all.

> [!NOTE]

> > Note that one can enable or disable certain [OpenSSL features] when building it, diverging from the default configuration. The Go runtime does not support all possible configurations, and some may cause the Go runtime to panic during initialization or not work as expected. The Go runtime is tested with the default configuration of the supported versions and with the OpenSSL configuration shipped in [Azure Linux].

Reviewer: What is 'it' referring to and do we start mention Az Linux 3 SSL impl here and throughout the doc?

Author: "It" is OpenSSL. I'm not sure AzL 3 needs any special mention other than maybe an entry in the changelog to note the Microsoft Go releases that include our compat changes for Azure Linux 3.
